stdlib/src/memory/unsafe_pointer.mojo

```diff
@@ -129,8 +129,11 @@ struct UnsafePointer[
 
     @staticmethod
     @always_inline
-    fn alloc(count: Int) -> Self:
-        """Allocate an array with default alignment.
+    fn alloc[alignment: Int = alignof[T]()](count: Int) -> Self:
+        """Allocate an array with specified or default alignment.
+
+        Parameters:
+            alignment: (Optional) The alignment in bytes of the allocated memory.
```

Comment:
**Suggestion** Remove "(Optional)" now that this is no longer literally an `Optional`.

Reply:
While the type isn't `Optional` it is optional for the user to specify it

I believe removing it would convey that the user is required to specify alignment

Though I'm open to removing it if you think we should after considering this

Reply:
Hi @sa-, your reasoning makes sense. This comes down to a matter of consistent style within stdlib code, and optional arguments and parameters are something we don't call out explicitly in general.

E.g. see this example of what we do for `print`:

![CleanShot 2024-06-13 at 08 55 50@2x](https://github.com/modularml/mojo/assets/5759631/ba3e9e9f-f9d9-443e-860d-518131734e33)

Reply:
Sounds good, i've removed it

Reply:
Thanks @sa-! 🙂

Reply:
Thanks to you as well for your time